### DISCUSSION TITLE
Use $data_source in format_item/format_album

### DISCUSSION THREAD

**Discussion by JOJ0** [beetbox/beets#4595] [Q&A] [answered]
Hi, a very basic question I can't seem to handle and been trying for a long time to get working already. From my understanding, this should be possible to do:
```
%if{$data_source,$data_source,ASIS}"
```

Above is part of my format_item and format_album config.

Why do I get outputs like that instead of falling back to the text ASIS:

```
beet ls data_source::'^$' sp-x
SP-X - Voltage EP - 00 - Suppressor Grid (Original Mix) (0000/)  [] MP3:320kbps a:11-12/$data_source
SP-X - Voltage EP - 00 - X-5684 (Original Mix) (0000/)  [] MP3:320kbps a:11-12/$data_source
```

data_source is empty on that release why doesn't it return False?

The first solution idea that comes up is that I should use %ifdef instead of %if, but no luck, it seems to always return False:

```
$ beet ls xhin                  
Xhin - Greyscale - 05 - Pages (2008/DE)  [MTIEFCD003] MP3:320kbps a:11-04/ASIS
Xhin - Drawer / Incidental (Remixes) - 03 - Incidental - Audio Injection & Drumcell Remix (0000/)  [] MP3:320kbps a:11-12/ASIS
Lucy, Xhin - LX2 / LX3 - 01 - LX2 (2011/Germany)  [CLR044] MP3:320kbps a:11-07/ASIS
```

```
$ beet ls xhin -f '$artist - $album - $title: $data_source'
Xhin - Greyscale - Pages: MusicBrainz
Xhin - Drawer / Incidental (Remixes) - Incidental - Audio Injection & Drumcell Remix: $data_source
Lucy, Xhin - LX2 / LX3 - LX2: Discogs
```

**Comment by wisp3rwind** [comment]
I was also going to suggest `%ifdef`, since `data_source` is a flexible attribute. How exactly did you use it? Note that the docs state

> %ifdef{field}, %ifdef{field,truetext} or %ifdef{field,truetext,falsetext}: Checks if an flexible attribute field is defined. If it exists, then return truetext or field (default). Otherwise, returns falsetext. **The field should be entered without $.**

**Comment by JOJ0** [comment, marked as answer]
Hi @wisp3rwind, thanks so much for the hint! That was the problem. Shame on me for not reading the docs. I _had_ read that actually a couple of days ago because of something else, but forgot again that I should remove the $.

Finally it's working. Correct usage is to check without the $, but add it when accessing:

```
%ifdef{data_source,$data_source,ASIS}"
```
